pghoard/config.py

```diff
@@ -23,16 +22,32 @@ def get_cpu_count():
     return multiprocessing.cpu_count()
 
 
-def find_pg_binary(program, versions=None):
+def find_pg_binary(wanted_program, versions=None):
+    if wanted_program in ('pg_receivexlog', 'pg_receivewal'):
+        programs = ['pg_receivexlog', 'pg_receivewal']
+    else:
+        programs = [wanted_program]
     pathformats = ["/usr/pgsql-{ver}/bin/{prog}", "/usr/lib/postgresql/{ver}/bin/{prog}"]
-    for ver in versions or SUPPORTED_VERSIONS:
+    versions = versions or SUPPORTED_VERSIONS
+    for ver in versions:
         for pathfmt in pathformats:
-            if LooseVersion(ver) >= "10" and program == "pg_receivexlog":
-                program = "pg_receivewal"
-            pgbin = pathfmt.format(ver=ver, prog=program)
-            if os.path.exists(pgbin):
-                return pgbin, ver
-    return os.path.join("/usr/bin", program), None
+            for program in programs:
+                pgbin = pathfmt.format(ver=ver, prog=program)
+                if os.path.exists(pgbin):
+                    return pgbin, ver
+    # We couldn't find a supported version in the "well known locations",
+    # let's search in the path.
+    for path in os.environ["PATH"].split(os.pathsep):
```

Comment:
Might be easier to use https://docs.python.org/3/library/shutil.html#shutil.which?

Reply:
sounds reasonable